apis/discovery/v1alpha1/foreigncluster_types.go

```diff
@@ -29,14 +29,29 @@ import (
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
+// JoinPhase indicates the phase of a peering with a remote cluster.
+type JoinPhase string
+
+const (
+	// JoinPhaseNone indicates that there is no peering.
+	JoinPhaseNone JoinPhase = "None"
+	// JoinPhasePending indicates that the peering is pending,
+	// and we are waiting fore the remote cluster feedback.
+	JoinPhasePending JoinPhase = "Pending"
+	// JoinPhaseEstablished indicates that the peering has been established.
+	JoinPhaseEstablished JoinPhase = "Established"
+	// JoinPhaseDeleting indicates that the peering is being deleted.
+	JoinPhaseDeleting JoinPhase = "Deleting"
```

Comment:
I would propose Disconnecting instead of Deleting, as you are not deleting an object, but disconnecting from a peer.